fvm/fvm_test.go

```diff
@@ -3696,6 +3702,66 @@ func TestSettingExecutionWeights(t *testing.T) {
+		fvm.WithAccountCreationFee(fvm.DefaultAccountCreationFee),
+		fvm.WithStorageMBPerFLOW(fvm.DefaultStorageMBPerFLOW),
+		fvm.WithExecutionMemoryLimit(
+			100_000_000,
+		),
+		fvm.WithExecutionMemoryWeights(
+			memoryWeights,
+		),
+	).run(
+		func(t *testing.T, vm *fvm.VirtualMachine, chain flow.Chain, ctx fvm.Context, view state.View, programs *programs.Programs) {
+			txBody := flow.NewTransactionBody().
+				SetScript([]byte(`
+				transaction {
+					prepare(signer: AuthAccount) {
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+						while true {break};while true {break};while true {break};while true {break};while true {break};
+					}
+				}
+			`)).
+				SetProposalKey(chain.ServiceAddress(), 0, 0).
+				AddAuthorizer(chain.ServiceAddress()).
+				SetPayer(chain.ServiceAddress())
+
+			err := testutil.SignTransactionAsServiceAccount(txBody, 0, chain)
+			require.NoError(t, err)
+
+			tx := fvm.Transaction(txBody, 0)
+			err = vm.Run(ctx, tx, view, programs)
+			require.NoError(t, err)
+			require.Greater(t, tx.MemoryUsed, uint64(100_000_000))
```

Comment:
This fails because the service account is no longer metered if they're the payer and `testutil.SignTransactionAsServiceAccount` sets `txBody.Payer` to the service account. It also does so for the proposal key so the related calls to `txBody` above aren't doing anything.